GitUpKit/Interface/GIGraphView.m

```diff
@@ -1495,46 +1515,27 @@ - (void)drawRect:(NSRect)dirtyRect {
   }
 #endif
 
-  // Find all lines in the drawing area
-  for (NSUInteger index = startIndex; index <= endIndex; ++index) {
-    GILayer* layer = layers[index];
-    for (GILine* line in layer.lines) {
-      if (![lines containsObject:line]) {
-        [lines addObject:line];
-      }
-    }
-  }
-
-  // Draw lines
-  if (lines.count) {
-    CGContextSetLineJoin(context, kCGLineJoinMiter);
-
+  // Draw all lines in the drawing area
+  do {
```

Comment:
Any particular purpose for the `do...while`?

Reply:
It's intended to demonstrate that the `SetBlendMode` calls are bracket the layers/lines enumeration. That said, it is stylistic, and thus removable. In my experience the bracketing of GState is a frequent source of bugs, though. If you'd rather not have the `do { … }`, I think I'd refactor out to a `drawLines` method instead.

Reply:
It's fine. Is there any difference between just having `{ ... }`? It's fine either way. I'll give this a test when I get the chance and probably merge it in.

Reply:
Almost certainly just me forgetting that syntax works in C. Good callout.